Core.ApplicationServices/Users/Write/UserWriteService.cs

```diff
@@ -82,18 +79,26 @@ public Result<User, OperationError> Update(Guid organizationUuid, Guid userUuid,
             }
 
             var user = userRes.Value;
-
             if (!_authorizationContext.AllowModify(user))
             {
                 return new OperationError(OperationFailure.Forbidden);
             }
 
-            var updateResult = PerformUpdates(user, organizationUuid, parameters);
+            var organizationResult = _organizationService.GetOrganization(organizationUuid);
```

Comment:
This method keeps growing and there are multiple results so it calls for a small rewrite. Using Match/Bind/Select methods will help.

First let's move the Organization call up (we want to scope the calls we have. If a user can't access an organization, he won't be able to get a user in the same organization). Then we do something like (you can improve some of the names):
`
        public Result<User, OperationError> Update(Guid organizationUuid, Guid userUuid, UpdateUserParameters parameters)
        {
            using var transactionManager = _transactionManager.Begin();

            var updateResult = _organizationService.GetOrganization(organizationUuid)
                .Bind(organization => _userService.GetUserInOrganization(organizationUuid, userUuid)
                    .Bind(CanModifyUser)
                    .Select(user => (organization, user))
                )
                .Bind(orgAndUser => PerformUpdates(orgAndUser.user, orgAndUser.organization, parameters)
                    .Select(user => (user, orgAndUser.organization)));

            if (updateResult.Failed)
            {
                transactionManager.Rollback();
                return updateResult.Error;
            }

            var user = updateResult.Value.user;
            _userService.UpdateUser(user, parameters.SendMailOnUpdate, updateResult.Value.organization.Id);
            transactionManager.Commit();
            return user;
        }

        private Result<User, OperationError> CanModifyUser(User user)
        {
            if (!_authorizationContext.AllowModify(user))
            {
                return new OperationError($"Not allowed to edit user with uuid: {user.Uuid}",
                    OperationFailure.Forbidden);
            }

            return user;
        }
`

Reply:
sorry, it didn't paste 100% well + move the new private method to the rest of private methods